Consider replacing hard-coded color value with variable #1355

erangeles opened the pull request:
@notmessenger not ready to be reviewed yet.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/softlayer/sl-ember-components/1413)
<!-- Reviewable:end -->

SpikedKira commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/styles/sl-select.less, line 479 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K8-lDVL89NT4VvK2A30:-K8-lDVL89NT4VvK2A31:452378863) ([raw file](https://github.com/softlayer/sl-ember-components/blob/937a9e6dcb90062bb1bb7d02827aa5b5745495f3/app/styles/sl-select.less#L479)):*
shouldn't this be @input-bg?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/styles/sl-select.less, line 479 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K8-lDVL89NT4VvK2A30:-K85EoOP_i08YinXsfym:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/937a9e6dcb90062bb1bb7d02827aa5b5745495f3/app/styles/sl-select.less#L479)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Merge conflict

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-:-K8LTct7V1M9xfsho2Er:-428090593)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
Reviewed 1 of 2 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

juwara0 commented:
@notmessenger - This looks ready to merge.

---

Reviewed 1 of 2 files at r3, 2 of 2 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-:-K8yjYL3m0O3rUaHCHYC:481157256)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 1 of 2 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---

*[app/styles/sl-grid.less, line 52 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Ii6-bG9vkDuO7ie_1:-K9Ii6-bG9vkDuO7ie_2:-1175537183) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-grid.less#L52)):*
For all of this color setting, if the background-color declaration was removed all together couldn't we achieve the same (that the `th` background color is the same as the table) without having to have all of these lines of declaration?

---

*[app/styles/sl-grid.less, line 173 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IiMIcnqgz-UeBmjro:-K9IiMIcnqgz-UeBmjrp:145037317) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-grid.less#L173)):*
Isn't this referring to the same table DOM structure (other than also having `.panel`)?  If so, why is `@panel-bg` being used when it is not used in the either and both `@table-bg` and `@body-bg` (with additional logic) are?

---

*[app/styles/sl-input.less, line 9 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IjEDnALehKD9zMNbZ:-K9IjEDnALehKD9zMNb_:1286894801) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-input.less#L9)):*
Wouldn't perhaps `@component-active-color` or `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 244 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkkQp6tMHNovk0kLs:-K9IkkQp6tMHNovk0kLt:-1404729187) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L244)):*
Wouldn't perhaps or `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 263 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkvXiIlb16rdmabha:-K9IkvXiIlb16rdmabhb:982871352) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L263)):*
Wouldn't perhaps `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 272 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkxeXtDXI2RX-ZaCw:-K9IkxeXtDXI2RX-ZaCx:982871352) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L272)):*
Wouldn't perhaps `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 318 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkzxB8Zg3Bn7nRrez:-K9IkzxB8Zg3Bn7nRrf-:1286894801) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L318)):*
Wouldn't perhaps `@component-active-color` or `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 430 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlOMUrZfM9mZvcOMy:-K9IlOMUrZfM9mZvcOMz:603500341) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L430)):*
Wouldn't perhaps `@component-active-color` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 437 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlLSA3hWsaQe1ubJc:-K9IlLSBG8q0WT5w4l1a:982871352) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L437)):*
Wouldn't perhaps `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 442 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlcNc4UvbkofMYZXj:-K9IlcNc4UvbkofMYZXk:-1856999680) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L442)):*
Wouldn't perhaps `input-bg-disabled` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 497 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Ilw9KK64PWoxhqDpg:-K9Ilw9KK64PWoxhqDph:982871352) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L497)):*
Wouldn't perhaps `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 528 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Im6UMQbEG_dVBZJfp:-K9Im6UMQbEG_dVBZJfq:1286894801) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L528)):*
Wouldn't perhaps `@component-active-color` or `@dropdown-bg` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 551 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Im9aRDJYTrDraBSwa:-K9Im9aRDJYTrDraBSwb:603500341) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L551)):*
Wouldn't perhaps `@component-active-color` be more appropriate, as they seem to be geared towards this UI effect?  @SpikedKira thoughts?

---

*[app/styles/sl-select.less, line 554 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9ImL_zXdCDUWPJRZgV:-K9ImL_zXdCDUWPJRZgW:-1224928788) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L554)):*
Wouldn't perhaps some other variables to be more appropriate for this UI effect?

---

*[app/styles/sl-select.less, line 557 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9ImQ3naPq9usItpvjP:-K9ImQ3naPq9usItpvjQ:-1224928788) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L557)):*
Wouldn't perhaps some other variables to be more appropriate for this UI effect?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
I'm looking into some of these other comments and will add to them shortly.

---

Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[app/styles/sl-grid.less, line 52 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Ii6-bG9vkDuO7ie_1:-K9SSPLGHlTIeEFk9MJY:44829447) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-grid.less#L52)):*
Nope.  When using the fixed header, the rows go behind the TH and therefor the TH must have a declared background so as not to see the row text behind the row headers.

---

*[app/styles/sl-grid.less, line 173 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IiMIcnqgz-UeBmjro:-K9SScA8a1WM9rAFE7D6:-1430195654) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-grid.less#L173)):*
Bootstrap has a panel bg variable that can be set.  We are trying to spoof our TH having a transparent background when it can not do so.  In order to do this we must say in effect: table header bg = (panel-bg ? panel-bg : (body-bg ? body-bg : white ) ).

---

*[app/styles/sl-select.less, line 442 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlcNc4UvbkofMYZXj:-K9STP7jgdS8gh8o4W-K:1790408293) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L442)):*
actually, there's a @text-muted to likely be used here

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-:-K9STrId-X_pvfojA6b_:-1054657901)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
Review status: all files reviewed at latest revision, 22 unresolved discussions.

---

*[app/styles/sl-select.less, line 244 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkkQp6tMHNovk0kLs:-K9SXu2oE0tN9Y9DGK5u:1771371147) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L244)):*
@dropdown-bg is a menu/navigation, not an input.  This one is correct.

---

*[app/styles/sl-select.less, line 263 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkvXiIlb16rdmabha:-K9SZitUHQqGPwSBIvjI:1461154452) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L263)):*
Let's maybe revert this one to a color.  I can not find any way to use this.  Seems to maybe be deprecated?

---

*[app/styles/sl-select.less, line 272 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkxeXtDXI2RX-ZaCw:-K9SaOHIz4WEe_YWbZwS:-80443881) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L272)):*
verified correct

---

*[app/styles/sl-select.less, line 318 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkzxB8Zg3Bn7nRrez:-K9SarazFk0afKZ2xFRP:1924907293) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L318)):*
there seems to be a bug where this state doesn't happen. regardless should be correct.

---

*[app/styles/sl-select.less, line 429 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SbTF-fFvBQ7jbGUNr:-K9SbTF-fFvBQ7jbGUNs:-2023221689) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L429)):*
@component-active-color

---

*[app/styles/sl-select.less, line 430 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlOMUrZfM9mZvcOMy:-K9SbUk3HDKGtEsJ8SJB:1611608033) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L430)):*
@component-active-bg

---

*[app/styles/sl-select.less, line 437 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlLSA3hWsaQe1ubJc:-K9ScJdPujRI6gTJkNEO:-527957756) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L437)):*
Unable to produce this state; this is theoretically fine.

---

*[app/styles/sl-select.less, line 497 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Ilw9KK64PWoxhqDpg:-K9SclptsKuSVOmSvp2Y:1471495601) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L497)):*
@input-bg is correct

---

*[app/styles/sl-select.less, line 528 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Im6UMQbEG_dVBZJfp:-K9Scyr47Cl3HdK35cbU:1533264152) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L528)):*
again, unable to produce this state; theoretically proper.

---

*[app/styles/sl-select.less, line 534 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SejJGAadK8EspvtFo:-K9SejJGAadK8EspvtFp:-894942236) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L534)):*
@btn-default-border

---

*[app/styles/sl-select.less, line 535 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SemwQhWSPCIbQTT6S:-K9SemwQhWSPCIbQTT6T:-1576309535) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L535)):*
@btn-border-radius-small

---

*[app/styles/sl-select.less, line 536 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SepByTV3flMu2oqiF:-K9SepByTV3flMu2oqiG:-2106160981) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L536)):*
@btn-default-color

---

*[app/styles/sl-select.less, line 547 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SeryTVpXWclU8Ba1a:-K9SeryTVpXWclU8Ba1b:926674996) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L547)):*
add 'background-color: @btn-default-bg' (should exist here, not sure if it got deleted)

---

*[app/styles/sl-select.less, line 549 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SfasN4s-U4mlWopLq:-K9SfasN4s-U4mlWopLr:-1672867781) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L549)):*
darken(@btn-default-bg, 10%) (taken from bootstrap/less/mixins/buttons.less)

---

*[app/styles/sl-select.less, line 550 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SfgRS0GVuv93h-Ime:-K9SfgRS0GVuv93h-Imf:-1068471298) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L550)):*
darken(@btn-default-border, 12%)

---

*[app/styles/sl-select.less, line 551 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Im9aRDJYTrDraBSwa:-K9Sda7ylXDZSIhEYQrC:-1178698753) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L551)):*
comment this line, or @btn-default-color

---

*[app/styles/sl-select.less, line 554 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9ImL_zXdCDUWPJRZgV:-K9SgWYRaFpRRmrsKlkL:526232294) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L554)):*
@text-muted

---

*[app/styles/sl-select.less, line 557 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9ImQ3naPq9usItpvjP:-K9SgcHI8s4jpqmJ1gQv:516810974) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L557)):*
@text-color

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
*[app/styles/sl-input.less, line 9 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IjEDnALehKD9zMNbZ:-K9SgjAsCGUHwL3eZ2DM:605456209) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-input.less#L9)):*
This should be correct.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 2 of 3 files reviewed at latest revision, 22 unresolved discussions.

---

*[app/styles/sl-select.less, line 263 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkvXiIlb16rdmabha:-K9T-mvrWBsG-HEMC7VU:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L263)):*
Done.

---

*[app/styles/sl-select.less, line 429 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SbTF-fFvBQ7jbGUNr:-K9T0CienbMCT8dkT7dG:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L429)):*
Done.

---

*[app/styles/sl-select.less, line 430 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlOMUrZfM9mZvcOMy:-K9T0D2BHJ2QBUuJoE3W:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L430)):*
Done.

---

*[app/styles/sl-select.less, line 442 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlcNc4UvbkofMYZXj:-K9T7DtAioyRAKD-EM5n:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L442)):*
Done.

---

*[app/styles/sl-select.less, line 534 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SejJGAadK8EspvtFo:-K9T6tERYqI_wvFnxeC3:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L534)):*
Done.

---

*[app/styles/sl-select.less, line 535 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SemwQhWSPCIbQTT6S:-K9T6ftD53n_esehIAJE:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L535)):*
Done.

---

*[app/styles/sl-select.less, line 536 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SepByTV3flMu2oqiF:-K9T6f_9cW4iih3N-wq9:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L536)):*
Done.

---

*[app/styles/sl-select.less, line 547 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SeryTVpXWclU8Ba1a:-K9T6BTBqdlcDgraZrup:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L547)):*
Done.

---

*[app/styles/sl-select.less, line 549 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SfasN4s-U4mlWopLq:-K9T5eOeTTEJNCG-f57S:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L549)):*
Done.

---

*[app/styles/sl-select.less, line 550 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SfgRS0GVuv93h-Ime:-K9T4b2J2o4kDtx_Ny4i:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L550)):*
Done.

---

*[app/styles/sl-select.less, line 551 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Im9aRDJYTrDraBSwa:-K9T2MBy9gbAssrQMeAF:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L551)):*
Done.

---

*[app/styles/sl-select.less, line 554 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9ImL_zXdCDUWPJRZgV:-K9T1yA8dhSVDQKwsHqY:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L554)):*
Done.

---

*[app/styles/sl-select.less, line 557 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9ImQ3naPq9usItpvjP:-K9T1xqP81ujAWVhqJw6:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L557)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
Review status: 2 of 3 files reviewed at latest revision, 16 unresolved discussions.

---

*[app/styles/sl-select.less, line 534 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SejJGAadK8EspvtFo:-K9TBYeSnr7LJy3G_uFN:-255448752) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L534)):*
Why did we drop the "1px solid" part?  Either put it back or use border style variables from bootstrap.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 2 of 3 files reviewed at latest revision, 16 unresolved discussions.

---

*[app/styles/sl-select.less, line 534 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9SejJGAadK8EspvtFo:-K9XfjEoYPB1f1Ok04mE:-499146009) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L534)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

juwara0 commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[app/styles/sl-input.less, line 9 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IjEDnALehKD9zMNbZ:-K9Z5M9wZe3ZI1Ba20xG:-2226706) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-input.less#L9)):*
Correct because it has the correct color value we want or correct contextual by what the variable is representing?  A menu is not an input, but more like the other two conceptually, in my mind.

---

*[app/styles/sl-select.less, line 244 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkkQp6tMHNovk0kLs:-K9Z5c6s30MpGqFls_Nm:-1564220124) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L244)):*
Select is not an input either.

---

*[app/styles/sl-select.less, line 437 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlLSA3hWsaQe1ubJc:-K9Z6I2E_5GzXjdHGg-g:2086214341) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L437)):*
Because it has the desired background color because of what the variables represent conceptually, because `@dropdown-bg` seems more conceptually on-point to me.

---

*[app/styles/sl-select.less, line 497 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Ilw9KK64PWoxhqDpg:-K9Z6ZhUcssOm3-b4atC:2086214341) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L497)):*
Because it has the desired background color because of what the variables represent conceptually, because `@dropdown-bg` seems more conceptually on-point to me.

---

*[app/styles/sl-select.less, line 528 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Im6UMQbEG_dVBZJfp:-K9Z6cKXxe1fAFBKQ_Xr:2086214341) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L528)):*
Because it has the desired background color because of what the variables represent conceptually, because `@dropdown-bg` seems more conceptually on-point to me.

---

*[app/styles/sl-select.less, line 429 \[r6\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Z646NfnOGy29mXX4R:-K9Z646Oyw3v_MwRlCcn:-1647481245) ([raw file](https://github.com/softlayer/sl-ember-components/blob/243f8f1d992d1a4f16359895cd8ea2095dbd4eb4/app/styles/sl-select.less#L429)):*
Shouldn't this be `@component-active-bg`?

---

*[app/styles/sl-select.less, line 430 \[r6\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Z69Pe0HbCO2B1rzUJ:-K9Z69Pfekq78E4RC_vx:-393316701) ([raw file](https://github.com/softlayer/sl-ember-components/blob/243f8f1d992d1a4f16359895cd8ea2095dbd4eb4/app/styles/sl-select.less#L430)):*
Shouldn't this be `@component-active-color`?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[app/styles/sl-input.less, line 9 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IjEDnALehKD9zMNbZ:-K9ZAGkA8EWGzcjQxfoB:186426531) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-input.less#L9)):*
I would argue that this is conceptually correct since the typeahead dropdown is an extension of the input itself.

---

*[app/styles/sl-select.less, line 244 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IkkQp6tMHNovk0kLs:-K9ZARZqw4sfqAAXN-Eg:1233785460) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L244)):*
In terms of bootstrap, the @input-*** variables apply to :input class elements, not just the input tagName

---

*[app/styles/sl-select.less, line 497 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Ilw9KK64PWoxhqDpg:-K9ZAml6ezaCZXOgwjkt:-2025008716) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L497)):*
In my mind, this is conceptually correct as per comments on the typeahead change in this same PR.

---

*[app/styles/sl-select.less, line 528 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Im6UMQbEG_dVBZJfp:-K9ZB-jU_73t_UKDyoqX:2042653267) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L528)):*
I believe this is conceptually correct in terms of bootstrap variables.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

SpikedKira commented:
*[app/styles/sl-select.less, line 437 \[r4\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9IlLSA3hWsaQe1ubJc:-K9ZBCy6VQD_M6vCvsAi:-1028672159) ([raw file](https://github.com/softlayer/sl-ember-components/blob/730aa0294d61af0ad12d22adb11c3bd83b543c8b/app/styles/sl-select.less#L437)):*
If I'm not mistaken, this is actually a UL that sits inside of the input field and holds a series of "cards" that look like buttons.  Those cards represent the currently selected items in a multi-select scenario.  Thus, being inside of the input itself should have the @input-bg color.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

erangeles commented:
Review status: 2 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[app/styles/sl-select.less, line 429 \[r6\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Z646NfnOGy29mXX4R:-K9ZCXc_xpdYCs6Jtu-A:1758359711) ([raw file](https://github.com/softlayer/sl-ember-components/blob/243f8f1d992d1a4f16359895cd8ea2095dbd4eb4/app/styles/sl-select.less#L429)):*
Fixed.

---

*[app/styles/sl-select.less, line 430 \[r6\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413#-K9Z69Pe0HbCO2B1rzUJ:-K9ZCXxEy1TApSGqSars:1758359711) ([raw file](https://github.com/softlayer/sl-ember-components/blob/243f8f1d992d1a4f16359895cd8ea2095dbd4eb4/app/styles/sl-select.less#L430)):*
Fixed.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

juwara0 commented:
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 1 of 1 files at r7.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/1413)*
<!-- Sent from Reviewable.io -->
